analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -255,25 +255,18 @@ private void VerifyDispose(MethodDeclarationSyntax disposeMethod, bool isSealedC
 
             private static bool HasArgumentValues(InvocationExpressionSyntax invocation, params string[] arguments) =>
                 invocation.HasExactlyNArguments(arguments.Length)
-                && invocation.ArgumentList.Arguments
-                             .Select((a, index) => a.Expression.ToString() == arguments[index])
-                             .All(matching => matching);
+                && invocation.ArgumentList.Arguments.Select((x, index) => x.Expression.ToString() == arguments[index]).All(x => x);
```

Comment:
This method is semantically wrong because it does not take named arguments into account. It works in this case because it is only called with a single argument. Consider renaming it to HasArgumentValue (without s) and replace `string[] arguments` with `string argument` to make clear, it is only working fo single arguments.

Reply:
`x.Expression.ToString()` looks suspicious. What it actually is trying to do is `semanticModel.GetConstantValue(x.Expression)`. It's not that bad in this case (the caller only passes "this", "true", and "false" and this shortcut performs quite well).

Consider replacing `x.Expression.ToString()` with `semanticModel.GetConstantValue(x.Expression)` (and some special handling for the "this" argument) and add a test case for `Dispose(falsyConst))`.

Reply:
That's out of scope for this PR. And would require adding test for this case.